apps/meteor/client/hooks/roomActions/useOutlookCalenderRoomAction.ts

```diff
@@ -1,15 +1,15 @@
-import { useSetting } from '@rocket.chat/ui-contexts';
+import { useUser } from '@rocket.chat/ui-contexts';
 import { lazy, useMemo } from 'react';
 
 import type { RoomToolboxActionConfig } from '../../views/room/contexts/RoomToolboxContext';
 
 const OutlookEventsRoute = lazy(() => import('../../views/outlookCalendar/OutlookEventsRoute'));
 
 export const useOutlookCalenderRoomAction = () => {
-	const enabled = useSetting('Outlook_Calendar_Enabled', false);
+	const user = useUser();
 
 	return useMemo((): RoomToolboxActionConfig | undefined => {
-		if (!enabled) {
+		if (!user?.settings?.calendar?.outlook?.Enabled) {
```

Comment:
This is a pretty long chain, it might worth moving this to it's own hook similar to `useUserId` or at least storing it in a variable to avoid repeating the chain. Same applies to the other files doing the same in this PR.